open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3818,176 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+
+    PlanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+        stats-fields:
+          description:
+            A list of fields that the client requests the server to send statistics
+            in each `FileScanTask` returned in the response
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+
+    TableScanContext:
+      anyOf:
+        - $ref: '#/components/schemas/SnapshotScanContext'
+        - $ref: '#/components/schemas/IncrementalSnapshotScanContext'
+
+    BaseTableScanContext:
```

Comment:
I don't think `BaseTableScanContext` is adding value here. It actually complicates the implementation, as I am start to implement the client-side change. We can move `type` to `TableScanContext`, and remove the extra layer between `TableScanContext` and `SnapshotScanContext`/`IncrementalSnapshotScanContext`.

Reply:
I think @rdblue also mentioned here https://github.com/apache/iceberg/pull/9695#discussion_r1688731466 for us to move these properties within the scan context to the preplan and plan table request models themselves.

Reply:
Yeah, I think fixing the additional level in the request objects will probably fix the issue here.